rs/sns/governance/canister/canister.rs

```diff
@@ -176,23 +169,39 @@ impl Environment for CanisterEnv {
 
     /// Return the canister's ID.
     fn canister_id(&self) -> CanisterId {
-        id()
+        CanisterId::unchecked_from_principal(PrincipalId::from(ic_cdk::id()))
     }
 
     /// Return the canister version.
     fn canister_version(&self) -> Option<u64> {
-        Some(dfn_core::api::canister_version())
+        Some(ic_cdk::api::canister_version())
+    }
+}
+
+fn now_nanoseconds() -> u64 {
+    if cfg!(target_arch = "wasm32") {
+        ic_cdk::api::time()
```

Comment:
NCR; 

I'm again wondering why this semantics isn't built into `ic_cdk::api::time` ...